src/mlpack/methods/softmax_regression/CMakeLists.txt

```diff
@@ -5,6 +5,7 @@ set(SOURCES
   softmax_regression_impl.hpp
   softmax_regression_function.hpp
   softmax_regression_function.cpp
+  softmax_main.cpp
```

Comment:
You shouldn't add the file here, but instead as its own executable, otherwise `softmax_main.cpp` will get compiled into `libmlpack.so`, which is not the right thing.  Try this instead:

```
add_executable(softmax_regression
  softmax_regression_main.cpp
)
target_link_libraries(softmax_regression
  mlpack
)
install(TARGETS softmax_regression RUNTIME DESTINATION bin)
```